python/api-examples-source/tutorials/execution-flow/fragments/tutorial-fragment-multiple-container.py

```diff
@@ -0,0 +1,56 @@
+import streamlit as st
+import time
+
+st.title("Cats!")
+
+row1 = st.columns(3)
+row2 = st.columns(3)
+
+grid = [col.container(height=200) for col in row1 + row2]
+safe_grid = [card.empty() for card in grid]
+
+
+def black_cats():
+    time.sleep(1)
+    st.title("🐈‍⬛ 🐈‍⬛")
+    st.markdown("🐾 🐾 🐾 🐾")
+
+
+def orange_cats():
+    time.sleep(1)
+    st.title("🐈 🐈")
+    st.markdown("🐾 🐾 🐾 🐾")
+
+
+@st.experimental_fragment
+def herd_black_cats(card_a, card_b, card_c):
+    st.button("Herd the black cats")
+    container_a = card_a.container()
+    container_b = card_b.container()
+    container_c = card_c.container()
+    with container_a:
+        black_cats()
+    with container_b:
+        black_cats()
+    with container_c:
+        black_cats()
```

Comment:
Is it intentional to do the extra steps rather than e.g.

```python
with card_a.container():
    black_cats()
```

(I think that's equivalent but if I'm getting that wrong, that would be a good reason lol)

Reply:
Ah, if you think the explanation is clearer with the way you have it currently, that's fine w me

Reply:
This is the most protective against undesired behavior when combined with empties. By defining the containers all first, a slow process prevents them from blanking out during a full rerun (which may or may not be desirable). The way it is has the ghosting/resolidifying effect on a full rerun. The compact form you suggested will clear the cards since there's a (possible) delay between creating the empty and the plain container.

(I also need to do some further tests and possibly report another fragment bug. I came across some unexpected behavior with some very subtle variations involving empties...)